Feature/get product detail

flutterindiadev opened the pull request:
Get product details and get product categories features added
add address usecase added
change address usecase added
delete address usecase added
get vouchers usecase added
verifyotp usecase added
fixed issues in entity lists

flutterindiadev committed:
get product details feature added

flutterindiadev committed:
get product detail feature added

flutterindiadev committed:
get product category feature added

flutterindiadev committed:
get categories feature added

naikdp7 commented on core/data/lib/src/repository/product_repository_impl.dart:
```diff
@@ -14,8 +14,13 @@ class ProductRepositoryImpl extends ProductRepository {
-      List<Product> products) async {
-    return await databaseProvider.saveProducts(products);
+  Future<Either<NetworkError, Product>> getProductDetail(int productId) async {
+    return await networkPort.getProductDetail(productId);
```
remove unnecessary await

flutterindiadev replied:
Fixed

naikdp7 commented on core/domain/lib/src/usecase/products/get_product_category_usecase.dart:
```diff
@@ -0,0 +1,26 @@
+  @override
+  Either<AppError, bool> verify() {
+    // TODO: implement verify
+    throw UnimplementedError();
```
return true instead of throwing an error if no verification needed

flutterindiadev replied:
Done

naikdp7 commented on core/domain/lib/src/usecase/products/get_product_detail_usecase.dart:
```diff
@@ -0,0 +1,28 @@
+  @override
+  Either<AppError, bool> verify() {
+    // TODO: implement verify
+    throw UnimplementedError();
```
same here

flutterindiadev replied:
Done

naikdp7 commented on infrastructure/network-retrofit/lib/src/models/product_category_response_entity.dart:
```diff
@@ -0,0 +1,43 @@
+  @override
+  ProductCategoryResponseEntity restore(List<ProductCategory> data) {
+    final List<ProductCategoryEntity> productCategoryResonseEntity =
+        List.generate(
```
please explain this

flutterindiadev replied:
This has been fixed

naikdp7 reviewed (changes requested):
Requested changes

flutterindiadev committed:
add address usecase added
change address usecase added
delete address usecase added
verifyotp usecase added
fixed issues in entity lists

flutterindiadev committed:
added all files to commit

flutterindiadev committed:
removed unnecessary await keyword
